cypress/integration/login.js

```diff
@@ -11,14 +11,15 @@ describe('Login functionality', function() {
         cy.get('[id="loginForm.userId"]').click({force:true})
         cy.get('[aria-label="Demo User"]').click()
         cy.get('[id="loginForm.role"]').click({force:true})
-        cy.get('[aria-label="User"]').click()
+        cy.get('[aria-label="Team Lead"]').click()
         cy.get('[type="submit"]').click()
 
         cy.url().should('include', '/time-logging')
         cy.get('.page__title').contains('Timesheets')
         cy.get('.calendar').should('be.visible')
         cy.get('.tile.form').should('be.visible')
         cy.get('.user-info__title').contains('Demo User')
-        cy.get('.main-nav').find('li').should('have.length', 1)
+        cy.get('.main-nav').find('li').should('have.length', 2)
+        cy.get(".calendar__date").contains(`${new Date().getDate()}`);
```

Comment:
.calendar__date selects all calendar slots, so your test will always pass, need to specify the locator to the specific calendar slot for today.